File: 0601_0700/0665/0665.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    h, m = (int(x) for x in f.read().split(':'))
    if m >= h % 10 * 10 + h // 10:
        h = (h + 1) % 24
    if h % 10 > 5:
        h += 10 - h % 10
    q.write(f"{h:02d}:{h % 10 * 10 + h // 10:02d}")

File: 0601_0700/0665/0665.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int h, m;

	fscanf(f, "%d:%d", &h, &m);
	if (m >= h % 10 * 10 + h / 10)
		h = (h + 1) % 24;
	if (h % 10 > 5)
		h += 10 - h % 10;
	fprintf(q, "%02d:%02d", h, h % 10 * 10 + h / 10);
	return 0;
}
